src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.h

```diff
@@ -227,13 +262,48 @@ struct FLTreeOnode final : Onode, Value {
     });
   }
 
+  void maybe_update_local_object_id(
+    NodeExtentMutable &payload_mut,
+    Recorder *recorder,
+    local_object_id_t id) {
+    auto &mlayout = *reinterpret_cast<onode_layout_t*>(
+      payload_mut.get_write());
+    local_object_id_t layout_id = mlayout.local_object_id;
+    if (layout_id == LOCAL_OBJECT_ID_NULL) {
+      mlayout.local_object_id = id;
+      if (recorder) {
+        recorder->encode_update(
+          payload_mut, Recorder::delta_op_t::UPDATE_LOCAL_OBJECT_ID);
+      }
+    } else {
+      ceph_assert(id == layout_id);
+    }
+  }
+
+  void maybe_update_local_clone_id(
+    NodeExtentMutable &payload_mut,
+    Recorder *recorder,
+    local_clone_id_t id) {
+    auto &mlayout = *reinterpret_cast<onode_layout_t*>(
+      payload_mut.get_write());
+    ceph_assert(id != LOCAL_CLONE_ID_ZERO);
+    mlayout.local_clone_id = id;
+    if (recorder) {
+      recorder->encode_update(
+        payload_mut, Recorder::delta_op_t::UPDATE_LOCAL_CLONE_ID);
+    }
+  }
+
   void mark_delete() {
     assert(status != status_t::DELETED);
     status = status_t::DELETED;
   }
 
-  laddr_t get_hint() const final {
-    return Value::get_hint();
+  laddr_hint_t get_hint(
+    std::optional<local_object_id_t> object_id,
+    std::optional<local_clone_id_t> clone_id,
+    bool is_metadata) const final {
+    return Value::get_hint(object_id, clone_id, is_metadata);
```

Comment:
Same here, use explict named methods other than a vague list of parameters, to be constrained to valid use cases only -- less error-prone and easy to understand.